pkgs/top-level/python-aliases.nix

```
@@ -0,0 +1,36 @@
+lib: self: super:
+
+with self;
+
+let
+  # Removing recurseForDerivation prevents derivations of aliased attribute
+  # set to appear while listing all the packages available.
+  removeRecurseForDerivations = alias: removeAttrs alias [ "recurseForDerivations" ];
```

The eval error was caused by this, sorry about that. Somehow blacked out and forgot that pythonPackages also contains functions and such.

```suggestion
  removeRecurseForDerivations = alias:
    if lib.isAttrs alias # removeAttrs throws in all other cases
    then builtins.removeAttrs alias [ "recurseForDerivations" ]
    else alias;
```

---

I used the variant that is used in aliases.nix because I don't fully understand this part to be honest.

---

The code from top level removes the attribute `recurseForDerivations` from an attribute set if it exists and is true. My code tries to always remove the `recurseForDerivations` attribute if it encounters an attribute set.

Shouldn't make a derifference in the grand scheme of things and only matters if we have sub attribute sets in pythonPackages anyways.